lib/core/diagnostics.js

```diff
@@ -114,6 +120,70 @@ if (undiciDebugLog.enabled || fetchDebuglog.enabled) {
   isClientSet = true
 }
 
+// Track fetch requests
+if (fetchDebuglog.enabled) {
+  const debuglog = fetchDebuglog
+
+  diagnosticsChannel.channel('undici:fetch:start').subscribe(evt => {
+    const {
+      input
+    } = evt
+    debuglog(
+      'fetch start event has started request to %s',
+      input
+    )
+  })
+
+  diagnosticsChannel.channel('undici:fetch:end').subscribe(evt => {
+    const {
+      input
+    } = evt
+    debuglog(
+      'fetch end event has received response from %s',
+      input
+    )
+  })
+
+  diagnosticsChannel.channel('undici:fetch:asyncStart').subscribe(evt => {
+    const {
+      input,
+      result,
+      error
+    } = evt
+    debuglog(
+      'async start event has received response for %s - HTTP %d, error is  %s',
+      input,
+      result.status,
+      error?.message
+    )
```

Comment:
asyncStart event has not always result?

Reply:
it should....

Reply:
![image](https://github.com/nodejs/undici/assets/5059100/ae21dddd-c686-4505-9486-83551b8b22c0)

Reply:
I have committed your suggestions, so will resolve this